app/questionnaire/questionnaire_store_updater.py

```diff
@@ -165,25 +165,32 @@ def remove_list_item_data(self, list_name: str, list_item_id: str) -> None:
 
     def _capture_dependencies_for_list_item_removal(self, list_name: str) -> None:
         """
-        If an item is removed from a list, dependencies of both the add block and remove block need to be captured
-        and their progress updated.
-        (E.g. dynamic-answers depending on the add block could have validation, so need revisiting if an item is removed)
+        If an item is removed from a list, dependencies of any child blocks need to be captured and their progress updated.
+        (E.g. dynamic-answers depending on a child block could have validation, so need revisiting if an item is removed)
         """
         if remove_block_id := self._schema.get_remove_block_id_for_list(list_name):
-            answer_ids = self._schema.get_answer_ids_for_block(remove_block_id)
-            for answer_id in answer_ids:
-                self._capture_block_dependencies_for_answer(answer_id)
+            self._capture_block_dependencies_for_all_answers_in_block(remove_block_id)
```

Comment:
Minor, we should move this below. I don't see a reason to handle remove like this and add/repeating differently. The one below covers multiple list collectors also.

Could be something like:
```py
child_blocks = (
    list_collector.get("add_block"),
    list_collector.get("remove_block"),
    *list_collector.get("repeating_blocks", []),
)

for child_block in child_blocks:
    if child_block:
        self._capture_block_dependencies_for_all_answers_in_block(
            child_block["id"]
        )
```